framework/src/wifi_manager/wifi_manager.c

```diff
@@ -489,284 +595,384 @@ wifi_manager_result_e wifi_manager_init(wifi_manager_cb_s *wmcb)
 
 	if (g_dw_nwevent_mqfd == (mqd_t)ERROR) {
 		ndbg("iotivity connect event message queue init fail");
-		goto error_with_mutex_destory_all;
+		return WIFI_MANAGER_FAIL;
 	}
 #endif
-
-	wifi_mutex_acquire(w_mutex, WIFI_UTILS_FOREVER);
-	result = wifi_utils_init();
-	if (result != WIFI_UTILS_SUCCESS) {
-		ndbg("wifi_utils_init fail");
-		goto error_with_mutex_release;
+	
+	wifi_utils_result_e wres = wifi_utils_init();
```

Comment:
Does wifi_utils_init() and wifi_utils_get_info(&info) need to be protected with LOCK_WIFIMNG?

Reply:
Yes it is